tests/test_precommit_setup.py

```
@@ -0,0 +1,197 @@
+        repo_root = Path(__file__).parent.parent
+        config_file = repo_root / ".pre-commit-config.yaml"
+
+        # Act
+        with open(config_file) as f:
+            config_data = yaml.safe_load(f)
+
+        # Assert - Extract all hook IDs from all repos
+        all_hooks = []
+        for repo in config_data["repos"]:
+            for hook in repo["hooks"]:
+                all_hooks.append(hook["id"])
```

Ruff (0.12.2)

Use `list.extend` to create a transformed list

(PERF401)